### PR TITLE
Fix Issue #875 - windows named pipes

### PR DESCRIPTION
**Pull request by jan-zajic**
This PR added support for Docker for Windows - https://docs.docker.com/docker-for-windows/install/ that supports only Windows 10 1607 Anniversary Update or later. This Docker in default configuration used Windows duplex named pipe.

### CODE REVIEW

**Comment by jan-zajic** [comment]
It would be nice to add Travis CI Job on Windows 10 and run integration tests that are in sources, but i do not know if it is possible.

File: src/main/java/com/spotify/docker/client/npipe/NamedPipeSocket.java [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+  }
+
+  @Override
+  public void bind(SocketAddress bindpoint) throws IOException {
+    throw new SocketException("Bind is not supported");
+  }
+
+  @Override
+  public InetAddress getInetAddress() {
+    return null;
```
Comment (davidxia) [inline src/main/java/com/spotify/docker/client/npipe/NamedPipeSocket.java]: Would throwing `UnsupportedOperationException` in these methods be clearer instead?

**Comment by davidxia** [comment]
Could you also update the CHANGELOG?

**Comment by mohitseth-nitrr** [comment]
@jan-zajic nicely done !!!

**Comment by jan-zajic** [comment]
I batlle tested some advanced operations, like coping files to container and execute. it gets stuck during copy tar archive upload. So i disable pooling for this type of connection. And when disabled everything workes fine. We have same problems and same solution at https://github.com/fabric8io/docker-maven-plugin, i forgot it first time.

**Pull request merged**